### PR TITLE
Add dockerfile

### PR DESCRIPTION
**Pull request by wileyj**
based off of https://github.com/blockstackpbc/neon-docker/blob/master/Dockerfiles/Dockerfile.p2p-firewall

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
+FROM rust as builder
+WORKDIR /src/bitcoin-p2p-firewall
+COPY . .
+RUN cargo install --target x86_64-unknown-linux-gnu --path /src/bitcoin-p2p-firewall
```
Comment (jcnelson) [inline Dockerfile:4]: Installing glibc should not be necessary.  It works just fine on musl.

Comment (wileyj) [reply Dockerfile:4]: thanks! this is what i was looking for. will update

Comment (wileyj) [reply Dockerfile:4]: updated in new commit

**Comment by jcnelson** [comment]
LGTM!

**Pull request merged**